tt_eager/tt_dnn/op_library/eltwise_unary/multi_core/eltwise_unary_op_multi_core_height_block_sharded.cpp

```diff
@@ -0,0 +1,171 @@
+    log_debug(LogOp, "input_tile_size: {}", input_tile_size);
+
+    auto src_buffer = input.buffer();
+    auto dst_buffer = output.buffer();
+
+    bool src_is_dram = src_buffer->buffer_type() == tt_metal::BufferType::DRAM ? 1 : 0;
+    TT_FATAL(src_is_dram == 0, "Input buffer should be in L1");
+    std::vector<uint32_t> reader_compile_time_args = {
+        in_cb_id,
+    };
+
+    bool dst_is_dram = dst_buffer->buffer_type() == tt_metal::BufferType::DRAM ? 1 : 0;
+    TT_FATAL(dst_is_dram == 0, "Output buffer should be in L1");
+
+    std::map<string, string> kernel_defines;
+
+    tt_metal::KernelHandle unary_reader_kernel_id = tt_metal::CreateKernel(
+        program,
+        "tt_eager/tt_dnn/kernels/dataflow/reader_unary_sharded.cpp",
+        all_cores,
+        tt_metal::ReaderDataMovementConfig(reader_compile_time_args, kernel_defines));
+
+    vector<uint32_t> compute_kernel_args_group_1 = {
+         num_tile_per_core, // per_core_block_cnt
+         1 // per_core_block_size
+    };
+
+    bool fp32_dest_acc_en = false;
+    bool math_approx_mode = std::all_of(op_chain.begin(), op_chain.end(), [](const auto& u) {return eltwise_unary_op_utils::get_op_approx_mode(u.op_type);});
+    std::map<string, string> unary_defines = eltwise_unary_op_utils::get_block_defines(op_chain);
+    auto eltwise_unary_kernel_group_1_id = tt_metal::CreateKernel(
+        program,
+        "tt_metal/kernels/compute/eltwise_sfpu.cpp",
+        all_cores,
+        tt_metal::ComputeConfig{
+            .math_fidelity = MathFidelity::HiFi4,
+            .fp32_dest_acc_en = fp32_dest_acc_en,
+            .math_approx_mode = math_approx_mode,
+            .compile_args = compute_kernel_args_group_1,
+            .defines = unary_defines
+        }
+    );
+
+    uint32_t start_input_stick_id = 0;
+    if(input.memory_config().memory_layout == TensorMemoryLayout::HEIGHT_SHARDED){
+        for (int32_t core = 0; core < ncores_nhw; ++core) {
+            CoreCoord core_coord(core % ncores_x, core / ncores_x); // logical
+            tt_metal::SetRuntimeArgs(
+                program,
+                unary_reader_kernel_id,
+                core_coord,
+                {
+                    (uint32_t)(num_tile_per_core),
+                }
+            );
+        }
+    } else if(input.memory_config().memory_layout == TensorMemoryLayout::BLOCK_SHARDED){
+        for (int32_t core_y = 0; core_y < ncores_nhw; ++core_y) {
+            for (int32_t core_x = 0; core_x < ncores_x; ++core_x) {
+                CoreCoord core_coord(core_x, core_y); // logical
+                tt_metal::SetRuntimeArgs(
+                    program,
+                    unary_reader_kernel_id,
+                    core_coord,
+                    {
+                        (uint32_t)(num_tile_per_core)
+                    }
+                );
+           }
+        }
+    }else{
+        TT_FATAL(false, "Only height and block memory is supported by this fuction");
+    }
```

Comment:
There is no need for all this code/loops if you are just supporting sharded->sharded?
Can't it just be replaced with

```
std::vector<uint32_t> reader_runtime_args = {(uint32_t)(num_tile_per_core)};
tt_metal::SetRuntimeArgs(program, unary_reader_kernel_id, all_cores, reader_runtime_args);
```